File: pdoo/ruby/weapontype.rb
#encoding:utf-8
module Deepspace
  # Tipos de armas
  module WeaponType
    class Type
      attr_reader :power

      def initialize(power_val)
        @power = power_val
      end
   end

    LASER = Type.new(2.0)
    MISSILE = Type.new(3.0)
    PLASMA = Type.new(4.0)
  end
end

File: pdoo/java/Deepspace/WeaponType.java
/**
 * @author Manuel Gachs Ballegeer
 */
package Deepspace;

/**
 * @brief Tipos de armas
 */
public enum WeaponType {
    LASER(2.0f),
    MISSILE(3.0f),
    PLASMA(4.0f);
    private float power;
    WeaponType(float powerValue) {power = powerValue;}
    float getPower() {return power;}
}